akka-stream-tests/src/test/scala/akka/stream/impl/TimeoutsSpec.scala

```diff
@@ -354,3 +357,24 @@ class TimeoutsSpec extends StreamSpec {
   }
 
 }
+
+class TimeoutChecksSpec extends WordSpecLike with Matchers {
```

Comment:
Added as a separate test because it's an implementation detail which has no relation to Akka or Akka Streams.